vector/src/test/java/im/vector/app/features/pin/lockscreen/fragment/LockScreenViewModelTests.kt

```diff
@@ -260,7 +263,44 @@ class LockScreenViewModelTests {
         events.assertValues(LockScreenViewEvent.AuthError(AuthMethod.BIOMETRICS, exception))
     }
 
-    private fun createViewState(
+    @Test
+    fun `when showBiometricPrompt handles isAuthDisabledError, canUseBiometricAuth becomes false`() = runTest {
+        val viewModel = LockScreenViewModel(createViewState(), pinCodeHelper, biometricHelperFactory, keysMigrator, versionProvider, keyguardManager)
+        val exception = BiometricAuthError(BiometricPrompt.ERROR_LOCKOUT_PERMANENT, "Permanent lockout")
+        coEvery { biometricHelper.authenticate(any<FragmentActivity>()) } throws exception
+
+        val events = viewModel.test().viewEvents
+        events.assertNoValues()
+
+        viewModel.handle(LockScreenAction.ShowBiometricPrompt(mockk()))
+
+        exception.isAuthDisabledError.shouldBeTrue()
+        events.assertValues(LockScreenViewEvent.AuthError(AuthMethod.BIOMETRICS, exception))
+        viewModel.test().states.assertLatestValue { !it.canUseBiometricAuth }
+    }
+
+    @Test
+    fun `when OnUIReady action is received and showBiometricPromptAutomatically is true it shows prompt`() = runTest {
+        // To force showBiometricPromptAutomatically to be true
+        every { biometricHelper.isSystemAuthEnabledAndValid } returns true
+        val viewModel = LockScreenViewModel(createViewState(), pinCodeHelper, biometricHelperFactory, keysMigrator, versionProvider, keyguardManager)
+        val events = viewModel.test().viewEvents
+        viewModel.handle(LockScreenAction.OnUIReady)
+        events.assertLatestValue(LockScreenViewEvent.ShowBiometricPromptAutomatically)
+    }
+
+    @Test
+    fun `when OnUIReady action is received and isBiometricKeyInvalidated is true it shows prompt`() = runTest {
+        // To force isBiometricKeyInvalidated to be true
```

Comment:
would lean towards replacing the comment with a named function as it'll be easy for the comments to become out of date/misaligned 

```kotlin
private fun givenBiometricKeyIsInvalidated() {
  every { biometricHelper.hasSystemKey } returns true
  every { biometricHelper.isSystemKeyValid } returns false
}
```

also has the benefit of keeping the test body smaller (and being reusable!) 

what do you think?

Reply:
Yes, that's a lot better. Thanks!